version: '3.9'

networks:
  app-tier:
    driver: bridge

services:

  db_exceptions:
    image: postgres
    restart: always
    shm_size: 128mb
    environment:
      PGUSER: postgre
      POSTGRES_USER: postgre
      POSTGRES_PASSWORD: postgre
      POSTGRES_DB: exceptions_journal
    networks:
      - app-tier
    ports:
      - "5332:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d postgres" ]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - local_pgdata:/var/lib/postgresql/data
  
  db_tree:
    image: postgres
    restart: always
    shm_size: 128mb
    environment:
      PGUSER: postgre
      POSTGRES_USER: postgre      
      POSTGRES_PASSWORD: postgre
      POSTGRES_DB: tree
    networks:
      - app-tier
    ports:
      - "5333:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d postgres" ]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - local_pgdata:/var/lib/postgresql/data
    
volumes:
  local_pgdata:

